.circleci/config.yml

```diff
@@ -233,39 +233,101 @@ jobs:
           path: upstream/paywall-preview-resources
       - install-sdkman
       - android/accept_licenses
-      - android-dependencies
-      - android/restore_build_cache
-      - run:
-          name: Verify purchases-android target SDK compatibility (currently 33)
-          command: ./gradlew :test-apps:testpurchasesandroidcompatibility:assembleDebug
+      - android/restore_gradle_cache
       - run:
-          name: Verify purchases-ui target SDK compatibility (currently 34)
-          command: ./gradlew :test-apps:testpurchasesuiandroidcompatibility:assembleDebug
+          name: Assemble all variants (warms Gradle cache once)
+          command: ./gradlew testClasses --parallel --no-daemon
+      - android/save_gradle_cache
+      - persist_to_workspace:
+          root: .
+          paths:
+            - .
+
+  test_defaults_debug:
+    <<: *android-executor
+    shell: /bin/bash --login -o pipefail
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - android/restore_gradle_cache
       - run:
-          name: Run Tests
-          command: ./gradlew lint test
+          name: Run Defaults Debug Tests
+          command: ./gradlew testDefaultsDebugUnitTest --parallel --no-daemon
       - run:
-          name: Consolidate artifacts
+          name: Collect JUnit XMLs
           command: |
-              mkdir -p build/test-results/
-              find . -type f -regex ".*/build/test-results/.*xml" -exec cp --parents {} build/test-results/ \;
+            mkdir -p build/test-results
+            find . -type f -regex ".*/build/test-results/.*xml" -exec cp --parents {} build/test-results/ \;
+      - store_test_results:
+          path: build/test-results
+      - store_artifacts:
+          path: build/reports
+
+  test_defaults_release:
+    <<: *android-executor
+    shell: /bin/bash --login -o pipefail
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - android/restore_gradle_cache
+      - run:
+          name: Run Defaults Release Tests
+          command: ./gradlew testDefaultsReleaseUnitTest --parallel --no-daemon
       - run:
-          name: Kover HTML
+          name: Generate Kover HTML
           command: ./gradlew purchases:koverHtmlReportDefaultsRelease
       - run:
-          name: Kover XML
+          name: Generate Kover XML
```

Comment:
So we're only using Kover for the defaults release flavor right? I guess this could leave some paths untested even if they are actually tested... but I don't have a better idea right now, so I'm ok with this 😅

Reply:
right! I noticed that the other day as well, that's mainly why I asked about revisiting it. Let's leave it as is since that's what we had yeah